### ISSUE TITLE
Expose Kernel Process Data

### ISSUE THREAD

**Issue by mlucool** [jupyterlab/jupyterlab#7478]
Moving this to a separate issues from https://github.com/jupyterlab/jupyterlab/pull/7252#issuecomment-549813603.

I propose we add the ability to understand more about the kernel itself from outside the process. For example, what is the host and pid? How much memory/cpu is it taking? What does the process tree look like?

This needs to be outside the kernel itself as it's very useful to look at when the kernel is doing work.

What do people think?

**Comment by mlucool** [comment]
Seems [adobe did something similar](https://medium.com/adobetech/reimagining-jupyter-notebooks-for-enterprise-scale-8bc6340d504a):

![image](https://user-images.githubusercontent.com/1813603/68767926-93032180-05ef-11ea-9515-37a40b6ea616.png)

While above is on the jupyter server, this is exactly the thing I think would be useful on each kernel

**Comment by dhirschfeld** [comment]
It would be very useful to see CPU / memory usage on a per-kernel basis in JupyterLab

**Comment by goanpeca** [comment]
Is this extension doing some of the requested idea already https://github.com/jtpio/jupyterlab-system-monitor ?

**Comment by mlucool** [comment]
I'm not aware of per-kernel option in jupyterlab-system-monitor, but I recall it being discussed elsewhere at some point.

cc @jtpio

**Comment by jtpio** [comment]
Yes there was a prototype PR in https://github.com/jtpio/jupyterlab-system-monitor/pull/36:

![kernel-metrics](https://user-images.githubusercontent.com/591645/85010827-f5616900-b160-11ea-8433-a16af5474cf1.gif)

If anyone want to pick it up, be my guest :)

There is also https://github.com/jupyter-server/jupyter-resource-usage/pull/41.

The feature should ideally go in https://github.com/jupyter-server/jupyter-resource-usage directly. At some point, we should also move the UI components from `jupyterlab-system-monitor` to `jupyter-resource-usage` to avoid confusion: https://github.com/jtpio/jupyterlab-system-monitor/issues/51

**Comment by echarles** [comment]
> The feature should ideally go in https://github.com/jupyter-server/jupyter-resource-usage directly. At some point, we should also move the UI components from jupyterlab-system-monitor to jupyter-resource-usage to avoid confusion: jtpio/jupyterlab-system-monitor#51

For now https://github.com/jupyter-server/jupyter-resource-usage looks like implementing the UI for both classic notebook (which I don't find the javascript source code in the repo...) and jupyterlab.

If we had more features to the lab, we just need to make sure the classic notebook behavior remains intact.

I also wonder how we make make remote kernels support also this feature, whereas in a enterprise gateway environment, or in the new kernel environment provisioning paradigm https://github.com/jupyter/jupyter_client/issues/608 ?
